feat: Added delete option in My PRs section

Doraemon012 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes #46
2. This PR does the following: 
    - Added a delete button in the MyPR section through which, users can delete their PRs.
    - The number of PRs and score of the user are also updated by the same.

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [ ] The PR is made from a branch that's **not** called "main/master".

Priyansh61 commented:
@Doraemon012 Since I cant login in the deploy preview can you add a video showing the changes you have made?

Doraemon012 commented:
@Priyansh61 Here's the video showing the changes that I have made. Please check this.


https://github.com/bsoc-bitbyte/BSoC-Website/assets/116624667/0fc25dd2-6322-4a8e-9184-902324c66977

### Changes made:
- Added a **Delete** button in the **My PRs** section.
- The button deletes the selected PR and updates the score and number of PRs of the user.
- Also added a **toast** which is displayed when a PR is deleted.

thejaysharma commented:
![Screenshot_20230713_131822](https://github.com/bsoc-bitbyte/BSoC-Website/assets/96112110/dc224e53-718c-4277-b85e-791a5a74f337)

@Doraemon012  Fix the navbar menu which is going upside.

Doraemon012 commented:
@thejaysharma Sir, I didn't change anything in the styles of navbar, and in MyPR.vue too, I just changed the number of columns to display the delete button. Here's the video of the original website:

https://github.com/bsoc-bitbyte/BSoC-Website/assets/116624667/77d055ac-bc7b-4ed9-8d12-39be0505af77

This bug is in the original website too, it is not because of my code. If you open a separate issue for this then, I can try to fix this in the same.

Priyansh61 commented on src/views/MyPR.vue:
```diff
@@ -37,13 +40,62 @@ import axios from "axios";
     components: {
         Nav,
     },
+
+    methods :
+    {
+        set_id (id){
+            return id;
+        },
+
+        delete_pr (id) {
+            this.formattedUserPRData.filter(async (doc) => {
+                
+                if (doc.id == id){
+                    const colRef = await projectFirestore.collection("dashboard-2023")
+                    const docRef = colRef.doc(doc.id)
+                    docRef.delete().then(() => {
+                        console.log(docRef + " deleted")
+                    }).catch((e) => {
+                        console.log(e.message)
+                    })
+                    var toast = "Your PR " + doc.message + " is deleted successfully."
+                    alert(toast)
+                    console.log("finished deleting")
+                    
+                    
+                
+            }})
+            
+        },
+
+        async update_score (uid, difficulty) {
+            
+            var current_score = 0
+            var new_score = 0
+            var current_prs = 0
```
@Doraemon012 Dont calll the methods in the `MyPR.vue` we have a seperate file for defining all the methods, see how we have defined it there and create your function in `src > composables. > deletedocuments.js`

We always try to refractor our code, so to have easy navigation and debugging, Therefore its a good practice to breakdown thingsj in different files instead of including them in a single file only.

Priyansh61 commented on src/views/MyPR.vue:
```diff
@@ -21,6 +21,9 @@
                     <div class="heading4 text-white">
                         <span>{{ user.time }}</span>
                     </div>
+                    <div>        
+                        <img @click="delete_pr(user.id), update_score(user.uid, user.difficulty)" :id="set_id(user.id)" src="../assets/delete-icon.svg" style="&:hover{background-color: aqua; cursor: pointer;}; height: 1.5rem; width: 1.5rem;" alt="delete">
+                    </div>
```
Also when the users presses on the delete button, show him a modal that once this is deleted, the action cant be undone, something like that and have options like continue or cancel because someone might accidentaly press the delete button.

So before deleting add a confirmation modal.

Priyansh61 commented on src/views/MyPR.vue:
```diff
@@ -37,13 +40,62 @@ import axios from "axios";
+                if (doc.id == id){
+                    const colRef = await projectFirestore.collection("dashboard-2023")
+                    const docRef = colRef.doc(doc.id)
+                    docRef.delete().then(() => {
+                        console.log(docRef + " deleted")
```
remove console logs after testing, its not a good practice to keep them there.

Priyansh61 commented on src/views/MyPR.vue:
```diff
@@ -37,13 +40,62 @@ import axios from "axios";
+                    const docRef = colRef.doc(doc.id)
+                    docRef.delete().then(() => {
+                        console.log(docRef + " deleted")
+                    }).catch((e) => {
+                        console.log(e.message)
+                    })
```
Show the message in a toaster message, when the deletion fails.

Priyansh61 commented on src/views/MyPR.vue:
```diff
@@ -37,13 +40,62 @@ import axios from "axios";
+                        console.log(e.message)
+                    })
+                    var toast = "Your PR " + doc.message + " is deleted successfully."
+                    alert(toast)
+                    console.log("finished deleting")
```
Instead of using a `alert` browser service to show the test message, can you make a new toaster component, which can be used throughout the website and matches with the site design.

Doraemon012 commented:
### Changes made:
1. Created a **Toast** component
    - A toast will be displayed when a PR is deleted.
2. Moved the deletePR function to deleteDocuments.vue

Doraemon012 commented:
@Priyansh61 Sir, please review this.

Priyansh61 reviewed (changes requested):
@Doraemon012 Impressive work! Few more nits...

1. Add the name to the column where the delete button resides, it is empty currently

![image](https://github.com/bsoc-bitbyte/BSoC-Website/assets/96513964/d67c573d-7860-4b74-868a-f10c632d3593)

2. If possible can you improve the styling of the toast further, and when it is successfull use green toast else use yellow/red for error. (suggestion).

And a suggestion, on Github the comments should be left for the maintainer to resolve, not the contributor, this way they can keep track if something needs to improved further or not.

Doraemon012 commented:
### Changes made:
- Added the heading for the column containing delete button.
- Styled the toast component


https://github.com/bsoc-bitbyte/BSoC-Website/assets/116624667/128bb59d-2b87-465a-beff-d86b128b4e34

Doraemon012 commented:
@Priyansh61 Sir, please check this.

Priyansh61 commented:
@Doraemon012 Awesome Job! LGTM!!

Priyansh61 commented:
@Doraemon012 Please create an issue for the navbar allignment.
